heron/proto/topology.proto

```diff
@@ -72,9 +78,11 @@ message Component {
   // The name of the component. Like Tail-FlatMap
   required string name = 1;
   // Only one of the below two are set
```

Comment:
let's clarify this comment to describe how what's set below, and it's meaning can be determined by the value of spec.